File: ems-frontend/src/services/PetService.js
import api from "./api";

const REST_API_URL = "/api/pets";

export const listPets = () => {
  return api.get(REST_API_URL);
};

export const searchPetsByName = (name) => {
  return api.get(`${REST_API_URL}/search`, { params: { name } });
};

export const createPet = (pet) => {
  return api.post(REST_API_URL, pet);
};

export const getPetById = (id) => {
  return api.get(REST_API_URL + "/" + id);
};

export const updatePet = (id, pet) => {
  return api.put(REST_API_URL + "/" + id, pet);
};

export const deletePet = (id) => {
  return api.delete(REST_API_URL + "/" + id);
};

export const adoptPet = (petId, studentId) => {
  return api.put(REST_API_URL + "/" + petId + "/adopt/" + studentId);
};

export const returnPetById = (petId) => {
  return api.put(REST_API_URL + "/" + petId + "/return");
};

File: ems-backend/src/main/java/net/fernandosalas/ems/service/PetService.java
package net.fernandosalas.ems.service;

import net.fernandosalas.ems.entity.Pet;

import java.util.List;

public interface PetService {

    Pet createPet(Pet pet);

    Pet getPetById(Long petId);

    List<Pet> getAllPets();

    List<Pet> searchPetsByName(String name);

    Pet updatePet(Long petId, Pet pet);

    Pet adoptPet(Long petId, Long studentId);

    Pet returnPet(Long petId, boolean incrementStudentReturnCount);

    void deletePet(Long petId);
}
